toolboxes/core/gpu/cuNDArray_blas.cu

```diff
@@ -217,40 +262,105 @@ namespace Gadgetron{
       throw std::runtime_error("Gadgetron::amin(): Invalid input array");
 
     int device = cudaDeviceManager::Instance()->getCurrentDevice();
-    int result;
-
-    CUBLAS_CALL(cublas_amin(cudaDeviceManager::Instance()->lockHandle(device),
-                            (int)x->get_number_of_elements(), x->get_data_ptr(), 1, &result));
+    size_t result;
+
+    // If number of elements in the array is greater than INT_MAX break it up and perform calculations this is done to
+    // support large data arrays
+    int num_splits = x->get_number_of_elements() / INT_MAX + 1;
+    int remainder = x->get_number_of_elements() - INT_MAX * (num_splits - 1);
+
+    for (int ii = 0; ii < num_splits; ii++) {
+      int interim_result;
+      CUBLAS_CALL(cublas_amin(cudaDeviceManager::Instance()->lockHandle(device),
+                              (ii == num_splits - 1) ? remainder : INT_MAX, // n number of elements
+                              x->get_data_ptr() + INT_MAX * ii, 1, &interim_result));
+      cudaDeviceManager::Instance()->unlockHandle(device);
+      if (ii == 0)
+        result = (size_t)interim_result - 1;
+      else if (((*x)[INT_MAX * ii + (size_t)interim_result - 1]) < ((*x)[result]))
+        result = INT_MAX * ii + (size_t)interim_result - 1;
+    }
+    
+    if( result > x->get_number_of_elements() ){
+      throw std::runtime_error("Gadgetron::amin(): computed index is out of bounds");
+    }
+    
+    return result;
+  }
+  
+  template<class T> size_t amin( cuNDArray<complext<T>>* x )
+  {
+    if( x == 0x0 )
+      throw std::runtime_error("Gadgetron::amin(): Invalid input array");
 
-    cudaDeviceManager::Instance()->unlockHandle(device);
+    int device = cudaDeviceManager::Instance()->getCurrentDevice();
+    size_t result;
+
+    // If number of elements in the array is greater than INT_MAX break it up and perform calculations this is done to
+    // support large data arrays
+    int num_splits = x->get_number_of_elements() / INT_MAX + 1;
+    int remainder = x->get_number_of_elements() - INT_MAX * (num_splits - 1);
+    complext<T> saved_value;
+
+    for (int ii = 0; ii < num_splits; ii++) {
+      int interim_result;
+      CUBLAS_CALL(cublas_amin(cudaDeviceManager::Instance()->lockHandle(device),
+                              (ii == num_splits - 1) ? remainder : INT_MAX, // n number of elements
+                              x->get_data_ptr() + INT_MAX * ii, 1, &interim_result));
+      cudaDeviceManager::Instance()->unlockHandle(device);
+      auto interim_value = (*x)[INT_MAX * ii + (size_t)interim_result - 1];
+      if (ii == 0)
+        result = (size_t)interim_result - 1;
+      else if ((abs(interim_value.real()) + abs(interim_value.imag())) <
```

Comment:
Is this logic correct? Is that how cublas determines min for a complex number?

Reply:
Also, I think you were right to define this comparison for the types (complex and not complex) in one place and then only have one implementation of this function. However, the comparison operators could be internal to this file (or even to the function).

Reply:
That would simplify the code but seems like it may not be needed and I can just take abs for both complex and real values similar to cuBlas. I had missed that the first time around but they take abs of real numbers as well for amin and amax (I don't know why - its killing the electrical engineer in me).